metadata-ingestion/src/datahub/ingestion/source/metabase.py

```diff
@@ -451,12 +467,16 @@ def get_datasource_urn(self, card_details):
                 .get("source-table")
             )
             if source_table_id is not None:
-                schema_name, table_name = self.get_source_table_from_id(source_table_id)
-                if table_name:
-                    source_paths.add(
-                        f"{f'{schema_name}.' if schema_name else ''}{table_name}"
-                    )
-        else:
+                if source_table_id.startswith('card__'):
+                    card_id = source_table_id.split('__')[1]
+                    card_model = card_details =  self.emit_card_id(card_id)
+                else:
+                    schema_name, table_name = self.get_source_table_from_id(source_table_id)
+                    if table_name:
+                        source_paths.add(
+                            f"{f'{schema_name}.' if schema_name else ''}{table_name}"
+                        )
+        elif query_type == "native" or card_model is not None:
```

Comment:
card_model will always be empty, do we need this card_model check?
What is the use of the card_model?

Reply:
Line 472. It is not always empty. Let's say the `source_table_id` start with 'card__', it will return a new card_model value. If `card_model` is not empty, the value should be the same for `card_details`

card_model is a new variable for indicator.

Reply:
You're right. My bad, let me change it.

Reply:
solved https://github.com/datahub-project/datahub/pull/6729/commits/fc1f85e551fceed7e57050b361f4ade8ad416753